runtime/error.c

```diff
@@ -1,9 +1,138 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
 #include "balrt.h"
+#include "third-party/libbacktrace/backtrace.h"
+
+#define INITIAL_FRAME_COUNT 1
+#define MAX_FRAME_COUNT 200
+#define MAX_BYTES_COUNT INT32_MAX
+#define INITIAL_BYTES_COUNT 8
+#define SKIP_FROM_END 1 // This number counts from the last stack frame
+#define THREAD 0
+
+typedef struct {
+    uint32_t nFrames;
+    uint32_t cFrames;
+    uint32_t nBytes;
+    uint32_t cBytes;
+    Frame *frames;
+    char *bytes;
+} FramesInfo;
+
+static void *memAlloc(uint64_t nBytes) {
+    void *p = malloc(nBytes);
+    if (p != 0)
+        return p;
+    fprintf(stderr, "out of memory\n");
+    fflush(stderr);
+    abort();    
+}
+
+static void *memReAlloc(void *ptr, uint64_t nBytes) {
+    void *p = realloc(ptr, nBytes);
+    if (p != 0) {
+        return p;
+    }
+    fprintf(stderr, "out of memory\n");
+    fflush(stderr);
+    abort();
+}
+
+static void on_error(void *data, const char *msg, int errnum) { 
```

Comment:
way is the name in caterpillar_case when above function is in camlCase. Neither is exposed to generated code, are they?

Reply:
Variable names of the function signature are the same as what is defined in libbacktrace. I have changed the function names to camalCase.